openshift: migrate to using buildGoPackage

vdemeester opened the pull request:
###### Motivation for this change

Part of #52469, make the `runc` derivation using `buildGoPackage` :angel: 

This means `openshift` becomes a multi-output derivation :
`openshift.bin` and `openshift`. (I'm not entirely sure on that one :sweat_smile: )

cc @Mic92 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Mic92 commented on pkgs/applications/networking/cluster/openshift/default.nix:
```diff
@@ -29,13 +29,15 @@ in stdenv.mkDerivation rec {
+  buildInputs = [ which rsync go-bindata kerberos clang ];
 
-  outputs = [ "out" ];
+  outputs = [ "out" "bin" ];
```
`bin` should become before `out`, but this is already the default so you can skip it.

Mic92 commented on pkgs/applications/networking/cluster/openshift/default.nix:
```diff
@@ -71,16 +74,12 @@ in stdenv.mkDerivation rec {
+    mkdir -p $bin/bin $out
+    cp -a "_output/local/bin/$(go env GOOS)/$(go env GOARCH)/"* "$bin/bin/"
     install -D -t "$out/etc/bash_completion.d" contrib/completions/bash/*
     install -D -t "$out/share/zsh/site-functions" contrib/completions/zsh/*
```
I would also put that in `$bin` so "${openshift}//share/zsh/site-functions" works as expected.

vdemeester replied:
ok :+1:

vdemeester committed:
openshift: migrate to using buildGoPackage

This means `openshift` becomes a multi-output derivation :
`openshift.bin` and `openshift`

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Mic92 commented:
@GrahamcOfBorg build openshift
